src/shared/Atlassian.Bitbucket/BitbucketHostProvider.cs

```diff
@@ -83,16 +90,56 @@ public async Task<ICredential> GetCredentialAsync(InputArguments input)
             {
                 throw new Exception("Unencrypted HTTP is not supported for Bitbucket.org. Ensure the repository remote URL is using HTTPS.");
             }
+        }
+
+        private async Task<ICredential> GetStoredCredentials(InputArguments input)
+        {
+            if (_context.Settings.TryGetSetting(BitbucketConstants.EnvironmentVariables.AlwaysRefreshCredentials, 
+                Constants.GitConfiguration.Credential.SectionName, BitbucketConstants.GitConfiguration.Credential.AlwaysRefreshCredentials, 
+                out string alwaysRefreshCredentials) && alwaysRefreshCredentials.ToBooleanyOrDefault(false))
+            {
+                _context.Trace.WriteLine($"Ignore stored credentials");
+                return null;
+            }
+
+            var targetUri = input.GetRemoteUri();
+            string credentialService = GetServiceName(input);
+            _context.Trace.WriteLine($"Look for existing credentials under {credentialService} ...");
+
+            var credentials = _context.CredentialStore.Get(credentialService, input.UserName);
+
+            if (credentials == null)
+            {
+                _context.Trace.WriteLine($"    Found none");
+                return null;
+            }
+
+            _context.Trace.WriteLineSecrets($"    Found credentials: {credentials.Account}/{{0}}", new object[] {credentials.Password});
+            
+            //check credentials are still valid
+            if (!await ValidateCredentialsWork(input, credentials, GetSupportedAuthenticationModes(targetUri))) {
+                return null;
+            }
+
+            return credentials;
+        }
+
+        private async Task<ICredential> GetRefreshedCredentials(InputArguments input)
+        {
+            _context.Trace.WriteLine($"Refresh credentials...");
+
+            var targetUri = input.GetRemoteUri();
 
             // Check for presence of refresh_token entry in credential store
-            string refreshTokenService = GetRefreshTokenServiceName(input);
+            var refreshTokenService = GetRefreshTokenServiceName(input);
 
             AuthenticationModes authModes = GetSupportedAuthenticationModes(targetUri);
 
             _context.Trace.WriteLine("Checking for refresh token...");
             ICredential refreshToken = SupportsOAuth(authModes) ? _context.CredentialStore.Get(refreshTokenService, input.UserName) : null;
             if (refreshToken is null)
             {
+                _context.Trace.WriteLine($"    Found none");
                 // There is no refresh token either because this is a non-2FA enabled account (where OAuth is not
                 // required), or because we previously erased the RT.
 
```

Comment:
GitHub doesn't let me leave a comment or suggestion on a "context" line... (grr)
Please can you remove this code (one line below) that is not used:

```csharp
// Check for the presence of a credential in the store
string credentialService = GetServiceName(input);
```